usersync/encoder_decoder_test.go

```diff
@@ -0,0 +1,70 @@
+package usersync
+
+import (
+	"net/http"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestEncoderDecoder(t *testing.T) {
```

Comment:
Round trip tests are good to ensure matching encoder / decoder functionality. I encourage you to the encode and decoder separately for special cases such as a nil cookie.